### PR TITLE
Dockerfile:Add arm64 support for building images

### PR DESCRIPTION
**Pull request by Jianlin-lv**
Add arm64 support for cilium-runtime and  cilium-builder

Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>

Related: #9898

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10618)

### CODE REVIEW

File: contrib/packaging/docker/Dockerfile.runtime [hunk 1]
```diff
@@ -61,15 +62,9 @@ make -j `getconf _NPROCESSORS_ONLN` && \
 strip bpftool && \
 cd ../../../../ && \
 #
-# bpf-map
-#
-curl -SsL https://github.com/cilium/bpf-map/releases/download/v1.0/bpf-map -o bpf-map && \
-chmod +x bpf-map && \
-strip bpf-map && \
-#
 # cni/loopback
```
Comment (aanm) [inline contrib/packaging/docker/Dockerfile.runtime:77]: I think @borkmann already removed this in master.

Comment (Jianlin-lv) [reply contrib/packaging/docker/Dockerfile.runtime:77]: I checked the master branch again and the bpf-map related code still exists. “removed” mean that bpf-map is no longer required to be installed in Dockerfile.runtime?

Comment (borkmann) [reply contrib/packaging/docker/Dockerfile.runtime:77]: The PR is still in progress as it was blocked by another one to be merged first: https://github.com/cilium/cilium/pull/10542

**Review by borkmann — APPROVED** [review]
Needs a rebase but looks generally okay to me.

**Comment by tklauser** [comment]
test-me-please

**Comment by tklauser** [comment]
test-with-kernel

**Review by tklauser — APPROVED** [review]

**Comment by aanm** [comment]
This PR is blocked until some changes are done by @borkmann upstream.

**Comment by joestringer** [comment]
@aanm @borkmann I think that the upstream Linux changes can be done in parallel, do we need to block Docker image building on that?

Also I think we still need to figure out how we can build & distribute builder images (we can distribute through quay.io / dockerhub if we push platform-dependent images, but not sure at this stage how to build them). But I believe that this workflow can also be worked on independent of this PR.

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**